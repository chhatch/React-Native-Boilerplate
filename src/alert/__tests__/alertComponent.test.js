import * as React from 'react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'

import AlertComponent from '../component'

const mockStore = configureStore([])

describe('Alert component', () => {
    it(`should render alerts if exist`, () => {
        store = mockStore({
            alert: {
                alerts: [
                    { id: 0, msg: 'test0', type: 'success' },
                    { id: 1, msg: 'test1', type: 'error' },
                ],
            },
        })
        const tree = renderer
            .create(
                <Provider store={store}>
                    <AlertComponent />
                </Provider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })

    it(`should not render alerts if none`, () => {
        store = mockStore({
            alert: {
                alerts: [],
            },
        })
        const tree = renderer
            .create(
                <Provider store={store}>
                    <AlertComponent />
                </Provider>
            )
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})
