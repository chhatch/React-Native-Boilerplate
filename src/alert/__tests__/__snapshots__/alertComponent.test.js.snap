// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert component should not render alerts if none 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#000",
      },
      undefined,
    ]
  }
/>
`;

exports[`Alert component should render alerts if exist 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
      },
      Object {
        "marginTop": 63,
        "position": "absolute",
        "width": 750,
        "zIndex": 999,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "backgroundColor": "green",
          "fontSize": 20,
        },
      ]
    }
  >
    test0
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#000",
        },
        Object {
          "backgroundColor": "green",
          "fontSize": 20,
        },
      ]
    }
  >
    test1
  </Text>
</View>
`;
